### ISSUE TITLE
TaxonomyPicker-Need a way to set termsetid dynamically

### ISSUE THREAD

**Issue by ashavibapat** [SharePoint/PnP#1562]
Thank you for reporting an issue or suggesting an enhancement. We appreciate your feedback - to help the team to understand your needs, please complete the below template to ensure we have the necessary details to assist you. If you have a actual question, we would ask you to use [SharePoint Developer Group](http://aka.ms/sppnp-community) at Microsoft Tech Community. Thanks!

####Which PnP repository should you use to report the issue?####
https://github.com/OfficeDev/PnP/tree/dev/Components/Core.TaxonomyPicker

#### Category
[ ] Enhancement

#### Environment
[ ] Office 365 / SharePoint Online
[ ] SharePoint 2013

If SharePoint on-premises, what's exact CU version: 

#### Expected or Desired Behavior
Enhancement:-
Need a way to reset the termset id after the taxpicker has been initialized or a way to delete/remove existing tax picker and initialize new one with new termset id.
It is beneficial in requirements where,we need to select different termsets depending on say a value selected in a ddl or radio buttons.

I tried doing something like this,without much luck.by adding one more method to TaxonomyPicker extension.

setNewTermset: function (control,options) {
            try {
                this.TermSet = new TermSet(options);               
                this.initialize(options.termSetId);
               // this.TermSet.initialize(options.termSetId);
            } catch (e) {
                ULSOnError("STACK : " + e.stack + " MESSAGE: " + e.message, document.location.href, 0);
            }
        }

**Comment by ashavibapat** [comment]
Hi,
I got it working by making below changes.The container divs and button are not added if already present .so it just initializes again with same divs and new termset id.
in TaxonomyPicker.initialize
  `  this._control = jQuery('<div class="cam-taxpicker"></div>');

            //detach the hidden field from the parent and append to the wrapper
            var parent = this._hiddenValidated.parent();
            if (!parent.hasClass('cam-taxpicker')) {
                this._hiddenValidated = this._hiddenValidated.detach();
                parent.append(this._control);
            }
            else {
                this._control = parent;
            }
            this._suggestionContainer = parent.find(".cam-taxpicker-suggestion-container").length > 0 ?parent.find(".cam-taxpicker-suggestion-container") :jQuery('<div class="cam-taxpicker-suggestion-container"></div>');
            if (!this._enterFillIn) {
                this._dlgButton = parent.find(".cam-taxpicker-button").length > 0 ? parent.find(".cam-taxpicker-button") : jQuery('<div class="cam-taxpicker-button"></div>');
            }
            if (!this._isReadOnly) {
                this._editor = parent.find(".cam-taxpicker-editor").length > 0 ? parent.find(".cam-taxpicker-editor") : jQuery('<div class="cam-taxpicker-editor" contenteditable="true"></div>');
                this._control.empty().append(this._editor).append(this._dlgButton).append(this._hiddenValidated);
                this._control.after(this._suggestionContainer);
            }
            else {

                this._editor = parent.find(".cam-taxpicker-editor-readonly").length > 0? parent.find(".cam-taxpicker-editor-readonly"): jQuery('<div class="cam-taxpicker-editor-readonly" contenteditable="false"></div>');
                this._control.empty().append(this._editor).append(this._hiddenValidated);
            }

            this._selectedTerms = new Array();
            this._editor.html('');`

**Comment by VesaJuvonen** [comment]
thx ashavibapat for sharing the solution for the enhancement idea as well.

**Comment by spdevdocs** [comment]
We closed this issue as it had not activity within last 180 days. This is a generic process we have decided to perform for issues, which have not been explicitly marked still to be "work in progress" based on tags. We are performing this cleaning to make sure that old issues that have already been solved (but not closed) or are no longer relevant are cleaned out and make the issues more manageable. If this issue still valid, we would ask you to **open a new issue** and follow the guidance in the issue template related on the recommended location. We do apologize any inconveniences this might cause. Please do remember that issues in the issue lists are also messages for others in the community, so you can also check if you can assist on any of them. *“Sharing is caring!”*
